### ISSUE TITLE
How to make hdf5 smaller

### ISSUE THREAD

**Issue by samwaseda** [pyiron/pyiron#808]
I'm having two problems right now:

 - SPHInX hdf5 files are for some reason extremely large (I'll look into this one myself)
 - Setting `job.project_hdf5['output'].remove_group()` removes the group, but somehow the hdf5 file is just as large as before.

I'd be glad if you could explain what's going on in one of the problems above. Thanks

**Comment by jan-janssen** [comment]
When you remove the group it is no longer indexed but it remains in the file, you can track the size of the HDF5 file using: 
`job.project_hdf5.get_size()` We implemented `job.project_hdf5.rewrite_hdf5()` to exclude groups and rewrite the file. @FeLoch introduced this last year https://github.com/pyiron/pyiron/pull/307

**Comment by jan-janssen** [comment]
Which Version of Sphinx are you using? the master branch? Can you compare the size of the sphinx files to the previous version?

**Comment by samwaseda** [comment]
> Which Version of Sphinx are you using? the master branch? Can you compare the size of the sphinx files to the previous version?

Master branch, but that of towards the end of June.

**Comment by jan-janssen** [comment]
Can you call the following on one of your HDF5 file?

```
h5 = h5py.File("demo.h5", 'r')

def printall(name, obj):
    print(name, dict(obj.attrs), len(obj))

h5.visititems(printall)
```

**Comment by samwaseda** [comment]
Okay I found the problem. I took one random example for which the hd5 was initially 239 M (while the output folder was 40 M). By erasing the following tags, I could get back:

 - `output/sphinx`: 123 M
 - `output/generic/dft/bands_eigen_values`: 54 M
 - `output/generic/dft/bands_occ`: 56 M

And in the end my hdf5 was 6 M. Eigen values and band occupancies probably shouldn't be removed right away from every simulation, but I guess I'm gonna remove `output/sphinx`, because it only stored the initial band occupancies etc. which I don't think anyone would ever need.
